src/core/CCList.ml

```diff
@@ -570,6 +572,29 @@ let combine_gen l1 l2 =
     res1 = res2)
 *)
 
+let combine_chop l1 l2 =
+  let rec direct i l1 l2 = match l1, l2 with
+    | (_, []) | ([], _) -> []
+    | _ when i=0 -> safe l1 l2 []
+    | (x1::l1', x2::l2') -> (x1, x2) :: direct (i-1) l1' l2'
+  and safe l1 l2 acc = match l1, l2 with
+    | ([], _) | (_, []) -> List.rev acc
+    | (x1::l1', x2::l2') -> safe l1' l2' @@ (x1, x2) :: acc
```

Comment:
```suggestion
    | (x1::l1', x2::l2') ->
      let acc = (x1,x2) :: acc in
      safe l1' l2' acc
```

(clearer what evaluation order is, and avoids the `@@`)

Reply:
Suits me. Actually I copied your own code for `combine` ;-)